sonarts-core/src/runner/processRequest.ts

```diff
@@ -37,7 +38,24 @@ export function processRequest(inputString: string) {
   const input = JSON.parse(inputString);
   let program = createProgram(input.tsconfig, input.projectRoot);
 
+  const filesNumber: number = input.filepaths.length;
+  let currentFile: string;
+  let processedFilesCounter = 0;
+  const currentTime = () => new Date().getTime() / 1000;
+  const logProgress = () => {
+    // log progress after 10s
+    if (Math.round(currentTime() - startTime) > 10) {
+      startTime = currentTime();
+      console.warn(
+        `${processedFilesCounter} files analyzed out of ${filesNumber}. Current file: ${basename(currentFile)}`,
```

Comment:
why `basename`? it will be too short, you might have files in different dirs with the same name